src/inference/dev_api/openvino/runtime/system_conf.hpp

```diff
@@ -131,9 +131,41 @@ OPENVINO_RUNTIME_API bool with_cpu_x86_avx512_core_amx_bf16();
  */
 OPENVINO_RUNTIME_API bool with_cpu_x86_avx512_core_amx();
 
+/**
+ * @brief      Checks whether cpu_mapping Available
+ * @ingroup    ie_dev_api_system_conf
+ * @return     `True` is CPU mapping is available, `false` otherwise
+ */
+OPENVINO_RUNTIME_API bool is_cpu_map_available();
+
+/**
+ * @brief      Returns number of CPU cores on Linux/Windows
+ * @ingroup    ie_dev_api_system_conf
+ * @return     A table about number of CPU cores of different types defined with ColumnOfProcessorTypeTable
+ * The following are two example of processor type table.
+ *  1. Processor table of two socket CPUs XEON server
+ *  ALL_PROC | MAIN_CORE_PROC | EFFICIENT_CORE_PROC | HYPER_THREADING_PROC
+ *     96            48                 0                       48          // Total number of two sockets
+ *     48            24                 0                       24          // Number of socket one
+ *     48            24                 0                       24          // Number of socket two
+ *
+ * 2. Processor table of one socket CPU desktop
+ *  ALL_PROC | MAIN_CORE_PROC | EFFICIENT_CORE_PROC | HYPER_THREADING_PROC
+ *     32            8                 16                       8           // Total number of one socket
+ */
+OPENVINO_RUNTIME_API std::vector<std::vector<int>> get_num_available_cpu_cores();
+
+/**
+ * @brief      Returns corresponding logical cores
+ * @ingroup    ie_dev_api_system_conf
+ * @param[in]  cpu_ids physical cores
+ * @return     logical cores corresponding to physical core.
+ */
+OPENVINO_RUNTIME_API std::vector<int> get_logic_cores(const std::vector<int> cpu_ids);
```

Comment:
```suggestion
OPENVINO_RUNTIME_API std::vector<int> get_logical_cores(const std::vector<int>& cpu_ids);
```

Reply:
@sunxiaoxia2022 please update it in the following flow refactoring PR.